### PR TITLE
Support install coreDNS

### PR DESCRIPTION
**Pull request by xxx7xxxx**
https://github.com/megaease/coredns is also updated (along with docker image) with the enhancements:

- Make DNSControlle be concurrently safe
- Dynamically retrieving easemesh control plane cluster IP instead of static config

### CODE REVIEW

**Review by localvar — APPROVED** [review]

**Comment by xxx7xxxx** [comment]
Update to use subcommand `emctl install coredns` to support install coredns alongside document.

File: docs/install.md [hunk 1]
```diff
@@ -245,19 +242,33 @@ If EaseMesh infrastructure has already been installed, the add-on features could
 emctl install --only-add-on --add-ons=ShadowService
 ```
 
+### Install CoreDNS
+
+If you want to support non-Java languages, installing EaseMesh dedicated CoreDNS is a must. It's very easy to do it:
```
Comment (zhao-kun) [inline docs/install.md:247]: non-Java Languages isn't correct, the pure spring-boot (without spring-cloud suite) application needs the CoreDNS installation also.

Comment (xxx7xxxx) [reply docs/install.md]: Updated

File: docs/install.md [hunk 2]
```diff
@@ -245,19 +242,33 @@ If EaseMesh infrastructure has already been installed, the add-on features could
 emctl install --only-add-on --add-ons=ShadowService
 ```
 
+### Install CoreDNS
```
Comment (zhao-kun) [inline docs/install.md:245]: I think it's necessary to remind users that the installation will overlap the system coredns image version and configuration.

File: emctl/cmd/client/command/meshinstall/coredns/deploy.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	return checkCoreDNSStatus(ctx.Client, ctx.Flags)
+}
+
+// PreCheck check prerequisite for installing mesh coreDNS
+func PreCheck(context *installbase.StageContext) error {
+	// Do nothing
+	return nil
+}
+
+// Clear clears all k8s resources about coreDNS
```
Comment (zhao-kun) [inline emctl/cmd/client/command/meshinstall/coredns/deploy.go:60]: We don't support revert installation of the coredns. Clear func should do nothing

File: docs/install.md (hunk 2 above)
Comment (xxx7xxxx) [reply docs/install.md:245]: Updated.

File: emctl/cmd/client/command/meshinstall/coredns/deploy.go (hunk 3 above)
Comment (xxx7xxxx) [reply emctl/cmd/client/command/meshinstall/coredns/deploy.go]: Updated

**Review by zhao-kun — APPROVED** [review]

**Pull request merged**